### PR TITLE
pl022 spi

### PR DESCRIPTION
**Pull request by vchong**
Depend on #896

### CODE REVIEW

**Comment by vchong** [comment]
\+ @daniel-thompson fyi only

File: core/drivers/pl022_spi.c [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+
+void pl022_end(void)
+{
+	/* disable ssp */
+	io_mask8(cfg->base + SSPCR1, SSPCR1_SSE_DISABLE, SSPCR1_SSE);
+}
+
+void pl022_init(const struct pl022_cfg *cfg_ptr)
+{
+	assert(cfg_ptr);
```
Comment (etiennec-linaro) [inline core/drivers/pl022_spi.c]: minor: could check here that global 'cfg' is not already set.

File: core/drivers/spi.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+/*
+ * Initialize the spi. The fields in the provided spi
+ * ops pointer must be valid.
+ */
+void spi_init(const struct spi_ops *ops_ptr)
+{
+	assert(ops_ptr);
+
+	ops = ops_ptr;
```
Comment (etiennec-linaro) [inline core/drivers/spi.c]: minor: same here, safeguard that 'ops' is not already set.

File: core/drivers/sub.mk [hunk 3]
```diff
@@ -2,11 +2,16 @@ ifeq ($(CFG_PL061),y)
 $(call force,CFG_GPIO,y,required by CFG_PL061)
 endif
 
+ifeq ($(CFG_PL022),y)
+$(call force,CFG_SPI,y,required by CFG_PL022)
```
Comment (etiennec-linaro) [inline core/drivers/sub.mk]: CFG_PL022 also requires GPIO support.

File: core/include/spi.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+void spi_txrx8(uint8_t *wdat, uint8_t *rdat, uint32_t num_txpkts, uint32_t *num_rxpkts);
+void spi_txrx16(uint16_t *wdat, uint16_t *rdat, uint32_t num_txpkts, uint32_t *num_rxpkts);
+void spi_tx8(uint8_t *wdat, uint32_t num_txpkts);
+void spi_tx16(uint16_t *wdat, uint32_t num_txpkts);
+void spi_rx8(uint8_t *rdat, uint32_t *num_rxpkts);
+void spi_rx16(uint16_t *rdat, uint32_t *num_rxpkts);
+void spi_init(const struct spi_ops *ops_ptr);
+void spi_test2(void);
```
Comment (etiennec-linaro) [inline core/include/spi.h]: minor: spi_test2() is not implemented.

**Comment by etiennec-linaro** [comment]
could the spi driver you propose be adapted to handle several SPI instances ?

File: core/drivers/pl022_spi.c (hunk 1 above)
Comment (vchong) [inline core/drivers/pl022_spi.c]: ok

File: core/drivers/spi.c (hunk 2 above)
Comment (vchong) [inline core/drivers/spi.c]: ok

File: core/drivers/sub.mk (hunk 3 above)
Comment (vchong) [inline core/drivers/sub.mk]: Ah, good catch. Thanks!

File: core/include/spi.h (hunk 4 above)
Comment (vchong) [inline core/include/spi.h]: Oops sorry.. leakage from test code. Will remove. Thanks!

File: core/drivers/pl022_spi.c [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <assert.h>
+#include <drivers/pl022_spi.h>
+#include <drivers/pl061_gpio.h>
```
Comment (vchong) [inline core/drivers/pl022_spi.c]: Remove this.

Comment (vchong) [inline core/drivers/pl022_spi.c]: ok

**Comment by vchong** [comment]
Closing. Will resubmit PR with multi-instance support.

**Comment by vchong** [comment]
Replaced by #905.

**Pull request closed without merging**